Sources/MapboxMaps/Location/Pucks/LocationPuckManager.swift

```diff
@@ -114,4 +128,14 @@ internal class LocationPuckManager: LocationConsumer {
             createPuck()
         }
     }
+
+    internal func changePuckBearingSource(to newPuckBearingSource: PuckBearingSource) {
+        puckBearingSource = newPuckBearingSource
+
+        if var puck = self.puck {
+            puck.puckBearingSource = newPuckBearingSource
+        } else {
+            createPuck()
+        }
+    }
```

Comment:
Would this not result in a puck always showing?

Reply:
yes that would, I think since puckBearingSource is a nonOptional, we should just omit the else statement here